src/rpc/rawtransaction.cpp

```diff
@@ -811,7 +811,7 @@ UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival
                 }
                 Coin newcoin;
                 newcoin.out.scriptPubKey = scriptPubKey;
-                newcoin.out.nValue = 0;
+                newcoin.out.nValue = MAX_MONEY;
```

Comment:
Please feel free to ignore this, but alternatively to express intent more directly,
```c++
newcoin.out.nValue = MAX_MONEY;
if (prevOut.exists("amount")) {
    newcoin.out.nValue = AmountFromValue(find_value(prevOut, "amount"));
}
```
can be replaced with a conditional expression:
```c++
newcoin.out.nValue = prevOut.exists("amount") ? AmountFromValue(find_value(prevOut, "amount")) : MAX_MONEY;
```